plugins/out_stackdriver/stackdriver.c

```diff
@@ -323,22 +327,68 @@ static int cb_stackdriver_init(struct flb_output_instance *ins,
     if (!token) {
         flb_plg_warn(ctx->ins, "token retrieval failed");
     }
-    if (ctx->metadata_server_auth) {
+
+    label = flb_kv_get_key_value("project_id", &ctx->resource_labels);
+    if (label == NULL) {
         ret = gce_metadata_read_project_id(ctx);
         if (ret == -1) {
+            flb_plg_error(ctx->ins, "get project_id failed");
             return -1;
         }
+    }
 
-        ret = gce_metadata_read_zone(ctx);
-        if (ret == -1) {
+    flb_plg_info(ctx->ins, "%s resource found", ctx->resource);
+
+    if (strncasecmp(ctx->resource, "gce_instance", 12) == 0) {
+        label = flb_kv_get_key_value("zone", &ctx->resource_labels);
+        if (label == NULL) {
+            ret = gce_metadata_read_zone(ctx);
+            if (ret == -1) {
+                flb_plg_error(ctx->ins, "get zone for gce_instance failed");
+                return -1;
+            }
+        }
+
+        label = flb_kv_get_key_value("instance_id", &ctx->resource_labels);
+        if (label == NULL) {
+            ret = gce_metadata_read_instance_id(ctx);
+            if (ret == -1) {
+                flb_plg_error(ctx->ins, "get instace_id for gce_instance failed");
+                return -1;
+            }
+        }
+    } else if (strncasecmp(ctx->resource, "generic_", 8) == 0) {
```

Comment:
minor change: put the ```else if``` in a new line:

ref: https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md#braces-usage-on-conditionals-loops-and-functions

Reply:
Done.